lib/generators/shopify_app/home_controller/home_controller_generator.rb

```diff
@@ -28,16 +27,26 @@ def add_home_index_route
 
       private
 
+      def embedded?
+        options['embedded'] == 'true'
+      end
+
       def embedded_app?
         ShopifyApp.configuration.embedded_app?
       end
 
-      def with_session_token?
-        @with_session_token
+      def with_cookie_authentication?
+        options['with_cookie_authentication']
       end
 
       def home_controller_template
-        with_session_token? ? 'unauthenticated_home_controller.rb' : 'home_controller.rb'
+        return 'unauthenticated_home_controller.rb' unless authenticated_home_controller_required?
```

Comment:
what's the difference between the `authenticated_home_controller_required?` flag and the `with_cookie_authentication?`. In what case would you have `with_cookie_authentication?` be false and also `authenticated_home_controller_required?` be false?